### ISSUE TITLE
volumeName/volumeID doesn't work when creating volume from existing one

### ISSUE THREAD

**Issue by pacoxu** [rexray/rexray#689]
volumeName Create from an existing volume ID. 
I use docker 1.13-rc4 and rexray 0.6.2, the scaleio is 2.0.0

According to doc below, we can use volumeName to create from existing volume.
![image](https://cloud.githubusercontent.com/assets/2010320/21678258/00bc7264-d379-11e6-81e9-0275290d9b4b.png)

I use the below command and opts are passed to docker volume correctly.
> [root@localhost ~]# docker volume create --driver rexray -o volumeName='rs1' -o volumeID='da93455600000000'  rs2
> rs2
> [root@localhost ~]# docker volume inspect rs2
> [
>     {
>         "Driver": "rexray",
>         "Labels": {},
>         "Mountpoint": "/",
>         "Name": "rs2",
>         "Options": {
>             "volumeID": "da93455600000000",
>             "volumeName": "rs1"
>         },
>         "Scope": "global",
>         "Status": {
>             "availabilityZone": "default",
>             "fields": null,
>             "iops": 0,
>             "name": "rs2",
>             "server": "scaleio",
>             "service": "scaleio",
>             "size": 16,
>             "type": "defaultSP"
>         }
>     }
> ]

However, the volume rs2 is not a snapshot of rs1.
![image](https://cloud.githubusercontent.com/assets/2010320/21678314/31922168-d379-11e6-8121-037c644c04f3.png)

I looked into the code https://github.com/codedellemc/libstorage/blob/master/drivers/storage/scaleio/storage/scaleio_storage.go#L349 and the volumeName and volumeID is not used

ScaleIO snapshot create should use something like /api/instances /System::{id}/action/snapshotVolumes

**Comment by cduchesne** [comment]
Thanks for pointing this out @pacoxu . This is a supported feature for drivers that support it. ScaleIO has no method of creating volumes from an existing snapshot. At least not that I've seen.

This does work with the Amazon EBS driver though:
```
[root@rexray-aws1 ~]# docker volume create --driver rexray --name vol01 --opt=snapshotID=snap-0da41ea35fe77b21a
vol01
[root@rexray-aws1 ~]# rexray volume ls
ID                     Name          Status       Size  Encrypted
vol-0d78fc2e47ded6d4a  vol01         available    16    false
```

**Comment by pacoxu** [comment]
@cduchesn I realized this. ScaleIO only support  to create a snapshot for an existing volume and create a snapshot for an existing snapshot.

Can rexray just create a snapshot when snapshotID or volumeID is in the option list?

Currently, I have to remove the ancestor of snapshot to use the snapshot. [I'm not sure whether my explanation is clear.  So here is an example.]
Example: There is a volume vol1 and I created snapshot snap1 for vol1. 
Then once the vol1 has some problem data, I want to revert to snap1. Firstly, at this point, I cannot use snap1 as it is not in the rexray volume list. I have to remove vol1, so that I can see snap1 from rexray. 

I have another idea about it. In my opinion, snapshot is just a volume type in ScaleIO like thin provisioned or thick provisioned. Why we don't take all scaleio volumes/snapshots as volumes. And we only need one option which would be volumeName/volumeID, we don't have to support snapshot ID/name. We can just create a snapshot for the existing volume/snapshot(which are same in user's view.) This is more like copying a volume than snapshot a volume in ScaleIO.

Best regards
Paco
